pykube/config.py

```diff
@@ -115,12 +115,17 @@ def users(self):
         Returns known users by exposing as a read-only property.
         """
         if not hasattr(self, "_users"):
-            us = {}
-            for ur in self.doc["users"]:
-                us[ur["name"]] = u = copy.deepcopy(ur["user"])
-                BytesOrFile.maybe_set(u, "client-certificate")
-                BytesOrFile.maybe_set(u, "client-key")
+            us = {
+                "default": {}
+            }
+
+            if "users" in self.doc:
```

Comment:
IIRC, based on the submitted kube config file, k8s can have a "user: default" which doesn't have any corresponding definition in users.  So I added this special value to accommodate that scenario.  It looks like recent comments below indicate you can also have a username that is empty with no other credentials.  So I figure we'd want to support all of these cases?